mksnapshot.sh

```diff
@@ -9,41 +9,43 @@
 
 echo "RUN: $0"
 if [ $# -lt 1 ]; then
-	echo "Usage: $0 [branch]"
-	echo "Where [branch] is the path under /bzr/ to the branch to snapshot."
+	echo "Usage: $0 [branch] [tag]"
+	echo "Where [branch] is the name of the branch to snapshot."
+	echo "Where [tag] is the name of the tag on that branch to snapshot."
 	exit 1
 fi
 # VCS details
-module=squid3
-BZRROOT=${BZRROOT:-/bzr}
-
-# generate a tarball name from the branch ($1) note that trunk is at
-# /bzr/trunk, but we call it 3.HEAD for consistency with CVS (squid 2.x), and
-# branches are in /bzr/branches/ but we don't want 'branches/' in the tarball
-# name so we strip that.
-branchpath=${1:-trunk}
-tag=${2:-`basename $branchpath`}
+if test "x${VCSROOT}" = "x"; then
+	echo "VCSROOT needs to be set to the base path for the repository to snapshot."
+	exit 1
+fi
+
+# generate a tarball name from the branch ($1) at tag ($2)
+branch=${1:-master}
+tag=${2:-HEAD}
 startdir=${PWD}
 date=`env TZ=GMT date +%Y%m%d`
 
-tmpdir=${TMPDIR:-${PWD}}/${module}-${tag}-mksnapshot
+tmpdir=${TMPDIR:-${PWD}}/squid-${tag}-mksnapshot/
 
 rm -rf ${tmpdir}
 trap "echo FAIL-BUILD_${VERSION} ; rm -rf ${tmpdir}" 0
+mkdir ${tmpdir}
 
 rm -f ${tag}.out
-bzr export ${tmpdir} ${BZRROOT}/${module}/${branchpath} || exit 1
+git checkout ${branch} || exit 1
+(git archive --format=tar ${tag} | tar -xC ${tmpdir}) || exit 1
 if [ ! -f ${tmpdir}/configure ] && [ -f ${tmpdir}/configure.ac ]; then
 	sh -c "cd ${tmpdir} && ./bootstrap.sh"
 fi
 if [ ! -f ${tmpdir}/configure ]; then
-	echo "ERROR! Tag ${tag} not found in ${module}"
+	echo "ERROR! Tag ${tag} not found."
 fi
 
 cd ${tmpdir}
-revision=`bzr revno ${BZRROOT}/${module}/${branchpath}`
+revision=`git rev-parse --short HEAD`
```

Comment:
Please see whether we can switch to using `git describe` instead. For example, I get the following output on your PR branch:

```
$ git describe --tags
SQUID_4_0_16-426-g56174e9
```

which is not usable "as is" but that could be because we do not have the right tag for the master branch (yet). I hope (but do not know for sure) that with the right setup you can replace all hard-coded branch names with `git describe`.